clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureResourceManagerClient.groovy

```diff
@@ -100,7 +90,8 @@ class AzureResourceManagerClient extends AzureBaseClient {
     ResourceGroupListParameters parameters = new ResourceGroupListParameters()
     parameters.setTagName("filter")
     parameters.setTagValue(applicationName)
-    return this.getResourceManagementClient(creds).getResourceGroupsOperations().list(parameters).resourceGroups
+
+    this.getResourceManagementClient(creds).getResourceGroupsOperations().list(parameters).resourceGroups
   }
 
   boolean resourceGroupExists(AzureCredentials creds, String resourceGroupName) {
```

Comment:
see line 61; we create a new resource group the first time we deploy a template if one does not exist already for the given application name and region